haystack/nodes/reader/table.py

```diff
@@ -128,27 +134,40 @@ def __init__(
         super().__init__()
 
         self.devices, _ = initialize_device_settings(devices=devices, use_cuda=use_gpu, multi_gpu=False)
-        config = TapasConfig.from_pretrained(model_name_or_path, use_auth_token=use_auth_token)
         if len(self.devices) > 1:
             logger.warning(
                 f"Multiple devices are not supported in {self.__class__.__name__} inference, "
                 f"using the first device {self.devices[0]}."
             )
 
-        if config.architectures[0] == "TapasForScoredQA":
-            self.model = self.TapasForScoredQA.from_pretrained(
-                model_name_or_path, revision=model_version, use_auth_token=use_auth_token
+        config = TapasConfig.from_pretrained(model_name_or_path, use_auth_token=use_auth_token)
+        self.table_encoder: Union[_TapasEncoder, _TapasScoredEncoder]
```

Comment:
Thanks for the suggestion!

Reply:
Unfortunately, I tried making the change, but it causes a mypy error because `_BaseTapasEncoder` doesn't have the attribute `model` which is used further down from here. So I reverted back to the original type declaration.

Reply:
I see, feel free to merge as soon as the CI has passed :)